src/str.c

```diff
@@ -195,12 +195,15 @@ char *mg_remove_double_dots(char *s) {
       while (s[0] != '\0') {
         if (s[0] == '/' || s[0] == '\\') {
           s++;
-        } else if (s[0] == '.' && s[1] == '.' &&
-                   (s[2] == '/' || s[2] == '\\')) {
-          s += 2;
-        } else {
-          break;
-        }
+        } else if (s[0] == '.' && s[1] == '.') {
+          if (s[2] == '\0' || s[2] == '/' || s[2] == '\\')
+            s += 2;
+          else 
+            break;
+          }
+          else {
+            break;
+          }
```

Comment:
Code-style: Whenever there is an `else` clause, use braces for both then and else paths. The else goes on the same line as the brace, short functions can be read without scrolling. Please run clang-format

Is there a reason I'm missing for the second `if` ? The `else` action is the same for both nested `if`s, why not keep the `&&` as before ?

If possible, I would add a test case in the unit tests, to make sure this works.